helix-term/src/commands.rs

```diff
@@ -5115,11 +5115,15 @@ fn surround_replace(cx: &mut Context) {
             }
         };
 
+        let selection = selection.clone();
+        let ranges: SmallVec<_> = change_pos.iter().map(|&pos| Range::point(pos)).collect();
+        doc.set_selection(view.id, Selection::new(ranges, 0));
```

Comment:
Okay, one last thing: you need to compute the correct primary_idx here or the view will be moved if you have a lot of selections: if you selected all the lines in a very big file and were viewing somewhere further down, an index of 0 would move the view all the way to the top

Something like

```rust
let primary = selection.primary();
let (i, _) = ranges.iter().enumerate().find(|_i, range| range.overlaps(primary))...

Selection::new(ranges, i)
```

Reply:
Good point. I've pushed [this change](https://github.com/helix-editor/helix/pull/7588/commits/6dd6347f732431fc78f5dd338572281ac4de3f9a) and so far so good during smoke testing,  but I wonder if there's a better way.